multiaddr/multiaddr.py

```diff
@@ -171,6 +178,29 @@ def protocols(self):
         """Returns a list of Protocols this Multiaddr includes."""
         return MultiAddrKeys(self)
 
+    def split(self, maxsplit=-1):
```

Comment:
I mostly just added it for symmetry with `str.split(sep=None, maxsplit=-1)`, but I could imagine that it could be used for decapsulation:

```py
import multiaddr

addr = multiaddr.Multiaddr("/ip4/127.0.0.1/tcp/8080/http/dns/example.com/tcp/443/https")
(target, endpoint, session) = addr.split(2)
# Use first two parts for creating an appropriate BSD socket and connecting
socket = establish_bsd_socket(multiaddr.Multiaddr.join(target, endpoint))
# Use remaining parts to expose a path-based mutliplexing protocol over the socket connection
# (in this case using HTTP CONNECT proxying, then TLS, then HTTP/?.?)
path_iface = establish_path_multiplexer(socket, session)
```

^ That is how I'd imagine (with more complexity baggage of course) how MutliAddr-native HTTP library would work internally.

Obviously there are also other ways to do this without the `maxsplit` parameter, but why not offer more options?

Reply:
Thanks a lot for the elaboration! I didn't notice there is `maxsplit` in `str.split`. No harm to have more options:)